pkg/server/server.go

```diff
@@ -378,15 +381,25 @@ func (h *handler) doMapping(identity *token.Identity) (string, []string, error)
 				return "", nil, fmt.Errorf("mapper %s renderTemplates error: %v", m.Name(), err)
 			}
 			return username, groups, nil
-		} else {
-			if err != mapper.ErrNotMapped {
-				errs = append(errs, fmt.Errorf("mapper %s Map error: %v", m.Name(), err))
-			}
-
-			if m.IsAccountAllowed(identity.AccountID) {
-				return identity.CanonicalARN, []string{}, nil
+		}
+		if h.userIDInUse {
+			//doMapping once more with UserID
+			mapping, err = m.Map(identity.UserID)
```

Comment:
Why do you do the ARN mapping AND the principal ID mapping?  Given there is an opt in switch, I'd think its one or the other.  In what case do we want both?

Reply:
I also don't understand how this is expected to work, you are overloading the parameter to Map which is supposed to be an ARN, and passing that to the RoleMapping lookup.  RoleMapping compares the ARN to the stored ARN loaded from the mapping.  I was expecting a new field for principal ID, which we can add to our structs in the Authenticator.

Reply:
What if we pass the entire `identity` to m.Map()? Then we can have logic that treats the ARN and the principal ID separately in the Map function based on whether principal ID mapping is enabled.

Reply:
thanks! I have updated by modifying the interface and passing the entire identity to m.Map